src/net.rs

```diff
@@ -31,6 +31,22 @@ pub trait NetworkListener: Clone {
     fn incoming(&mut self) -> NetworkConnections<Self> {
         NetworkConnections(self)
     }
+
+    /// Sets the read timeout for all streams that are accepted
+    fn set_read_timeout(&mut self, _: Option<Duration>) {
+        // This default implementation is only here to prevent the addition of
+        // these methods from being a breaking change. They should be removed
+        // when the next breaking release is made.
+        unimplemented!();
```

Comment:
Since these methods are called unconditionally later, anyone just upgrading hyper will find that suddenly, their custom `NetworkConnector` panics, which would be a breaking change. These should probably just do nothing, instead of calling `unimplemented!`.

If we want, we could put a `warn!` or even `error!` in here instead, so people looking at logs can see they should implement these. Or we could just do nothing.